mist-core/src/main/java/edu/snu/mist/core/task/groupaware/GroupAwareQueryManagerImpl.java

```diff
@@ -191,6 +193,10 @@ public QueryControlResult createQueryWithCheckpoint(final AvroDag avroDag,
       }
 
       final ApplicationInfo applicationInfo = applicationMap.get(appId);
+      if (applicationInfo.getGroups().size() == 0) {
+        createGroup(applicationInfo);
```

Comment:
Multiple groups can be created at the same time because several threads can execute the this code.

Reply:
What do you think adding a synchronized block (on `applicationInfo`) around the if statement(lines 196~198)?

Reply:
I've decided to use another method.